DirectEmployers/MyJobs
# Tag jobs with matching facets on import, then search by those tags.

~~Turns out custom facets are poorly tested, so I want to get more coverage on those.~~ This is done.
~~I also want to squash all the new migrations down into a single migration.~~ Done
~~@mklauber and I also need to fix the inf required for tests to pass.~~ They pass.
~~Other than that~~ this should be ready for review.

## What's this PR do?
What the PR title says.

## Screenshots (if appropriate)
Everything should look the same as it does now.

## Where should the reviewer start?
Imports and tag.py is a pretty logical place.
After that, most of the search logic has changed in helpers.py although search_views.py

## How should this be manually tested?
Pick a company site and associated business unit. If you don't have one already set up and want help setting one up I can help.
There are three toggles you need to flip for whatever business units/sites you want to test:

1. BusinessUnit.tag_facets needs to be True
2. SeoSite.tag_facets needs to be True
3. Configuration.facet_type needs to be Configuration.FACET_TAGS

If anyone is interested about the logic behind flags 1 & 2, the logic in tags.py should explain why, but I can also try to explain it if that doesn't make sense.

After you've done that, run imports for whatever buid/site you chose, and then browse around that site. Everything should look the same.

Specifically: auto-complete, facets on the right hand column, search results after navigating to a facet, the site map, and the rss feed should all be the same.

## Any background information you want to provide?
Some notable commits that don't actually change much:
7977b2dbf1f4d8b1a9e396995f339e2e209cbda7 - a refactor of `get_jobs` that doesn't change it's functionality, but does make it much easier to work with.
e5fd0e2bfa5f9fee723af459cbe693d8c0a52445 - moving tests that need tested on the new faceting method to a place where they're more reusable
9753b468177814c729b5e09094740734b4920353 - same as above
